Source/Core/Core/HW/SystemTimers.cpp

```diff
@@ -109,10 +109,26 @@ int GetAudioDMACallbackPeriod()
          (Mixer::FIXED_SAMPLE_RATE_DIVIDEND * 4 / 32);
 }
 
+static TimePoint next_audio_update(DT::zero());
+
 void AudioDMACallback(Core::System& system, u64 userdata, s64 cyclesLate)
 {
-  DSP::UpdateAudioDMA();  // Push audio to speakers.
-  system.GetCoreTiming().ScheduleEvent(GetAudioDMACallbackPeriod() - cyclesLate, et_AudioDMA);
+  const TimePoint now = Clock::now();
+
+  if (now > next_audio_update)
+  {
+    DSP::UpdateAudioDMA();  // Push audio to speakers.
+
+    DT max_variance =
+        std::chrono::duration_cast<DT>(DT_ms(Config::Get(Config::MAIN_TIMING_VARIANCE)));
+    if (now - next_audio_update > max_variance)
+      next_audio_update = now;
+
+    next_audio_update +=
+        std::chrono::duration_cast<DT>(DT_s(GetAudioDMACallbackPeriod()) / GetTicksPerSecond());
+  }
+
+  system.GetCoreTiming().ScheduleEvent(GetAudioDMACallbackPeriod() / 60 - cyclesLate, et_AudioDMA);
```

Comment:
So this triggers audio events 60x as often, which translates to a max potential audio speedup of 60x. 

We might want to limit this to a more sane value of say 10x. As larger values are more likely to trigger bugs in the game.

Reply:
yeah that sounds about right, this took like 5 seconds to throw together so it was more of a POC

Reply:
We can also shedule async events from other threads, however they might get delayed a bit longer. Nevertheless, they should kind of run after a given amount of real time.